src/lib/consensus/proof_of_stake.ml

```diff
@@ -392,7 +410,14 @@ module Data = struct
             s old.genesis_epoch_snapshot
             (* assume these keys are different and therefore we haven't checked any
          * slots or epochs *)
-        ; last_checked_slot_and_epoch= (Epoch.zero, Epoch.Slot.zero)
+        ; last_checked_slot_and_epoch=
+            make_last_checked_slot_and_epoch_table
+              (Public_key.Compressed.Table.create ())
+              proposer_public_keys
+              ~default:
+                ((* TODO: Be smarter so that we don't have to look at the slot before again *)
+                 let epoch, slot = Epoch.epoch_and_slot_of_time_exn now in
+                 (epoch, UInt32.(if slot > zero then sub slot one else slot)))
```

Comment:
This would have the side effects of skipping or redoing slots for existing entries, right?

Reply:
as discussed, it should work